### ISSUE TITLE
Possibility to revoke auth via this library

### ISSUE THREAD

**Issue by unreal4u** [mollie/oauth2-mollie-php#20]
We have already integrated Mollie OAuth (onboarding flow) but we want to integrate the revoke call as well as we think it might help with the confusion some customers have with the different profiles:
https://docs.mollie.com/reference/oauth2/revoke-token

However, there is no call in this library to do so. Can this be implemented so that we can initiate the call? Should this be implemented in this library or this one? 
https://github.com/mollie/mollie-api-php

Greetings.

**Comment by sandervanhooft** [comment]
I'd say implement it here on the oAuth client. What do you think @MollieRick ?

**Comment by MollieRick** [comment]
> I'd say implement it here on the oAuth client. What do you think @MollieRick ?

Yes - seems more logical to do it here!

**Comment by sandervanhooft** [comment]
@unreal4u Could you submit a PR?

**Comment by sandervanhooft** [comment]
Hi @unreal4u,

I've prepared a PR here: #21 , pending review from Mollie. Feel free to check it out and let me know if it works for you.

**Comment by unreal4u** [comment]
Thanks @sandervanhooft ! I have had no time available yet to begin developing this, so thanks a lot! I will be able to test this next week and will let you know!

**Comment by sandervanhooft** [comment]
Thanks @unreal4u, looking forward to your response.

**Comment by FawadNL** [comment]
Hi guys, I know this is not the place to ask but not sure where to ask and its quite urgent. Is the access_token and refresh_token user bound? Right now I am assuming its user bound:

`update_user_meta($userid, '_mollie_connect_access_token', $access_token);`
`update_user_meta($userid, '_mollie_connect_refresh_token', $refresh_token);`
`update_user_meta($userid, '_mollie_connect_expiry_time', $expire_time);`

But if the tokens are applicable for all merchants then I can save them as global variables and remove a lot of code and fix the token refresh issue in one go. 
@sandervanhooft @MollieRick

**Comment by sandervanhooft** [comment]
The client id and client secret are app-bound (so "global" in your application). The whole idea behind the OAuth flow is that you obtain the credentials that allow you to act on the merchant's behalf. So yes, the tokens are related to the user, not global. You can learn more [here](https://docs.mollie.com/oauth/overview).

**Comment by FawadNL** [comment]
Thanks for clarifying it @sandervanhooft ! Ran into refreshToken issue and eventually got everything working on WordPress. Calling this object is not working on WP:

`$grant = new \League\OAuth2\Client\Grant\RefreshToken();`

Instead, I replaced it with its actual value like this: 
`$token = $provider->getAccessToken('refresh_token', ['refresh_token' => $refreshToken]);`

That solved the issue. The error I received was "Fatal Uncaught Error: unknown argument $grant passed". 
Am still not sure why, cause autoload.php is being loaded, confirmed via logging.
